tcti/tcti_socket.c

```diff
@@ -25,6 +25,7 @@
  * THE POSSIBILITY OF SUCH DAMAGE.
  **********************************************************************/
 
+#define _GNU_SOURCE
```

Comment:
`limits.h` will declare `HOST_NAME_MAX` when `_POSIX_C_SOURCE` is defined. `_GNU_SOURCE` is a superset of `_POSIX_C_SOURCE` and we don't need all of the extra stuff that comes with it. Feature test macros should be defined project-wide through the build environment. There are two examples here: https://github.com/intel/tpm2-tss/blob/master/configure.ac#L60